### PR TITLE
Text input group guidelines

### LINKED ISSUE #2759: Create guidelines for text input group component
Guidelines for this demo basically: https://www.patternfly.org/v4/components/text-input-group/react-demos
Original issue + background here: https://github.com/patternfly/patternfly-design/issues/1088

Would add design guidelines to [text input group component](https://www.patternfly.org/v4/components/text-input-group). 

The text input group component essentially does the same thing as an attribute-value filter, except that a user can type it all in/select attribute value pairs inside of a text input.

### PR DESCRIPTION
**Pull request by mcarrano**
Adds design guidelines for the Text input group components.

Closes #2759

### CODE REVIEW

**Comment by mcarrano** [comment]
@mmenestr looks like I've sorted out the issues with this PR. Please review when you have a chance.

File: packages/v4/patternfly-docs/content/design-guidelines/components/text-input-group/text-input-group.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+2. **Input field:** The input field itself can be customized to use multiple text styles as shown here. This styling is not possible using a standard text input.
+
+3. **Clear button (optional):** A clear button can be added to clear the contents of the field.
+
+4. **Chip group (optional):** A chip group can be nested inside of the text input as shown here.
+
+## Usage
+
+The text input group component is useful for building more complex custom inputs. It's usage includes, but is not limited to, the following use cases. Each of these can be found in a [React demo here](/components/text-input-group/react-demos).
```
Comment (mmenestr) [inline packages/v4/patternfly-docs/content/design-guidelines/components/text-input-group/text-input-group.md:22]: Typo - should be "its" usage, not "it's" usage

Comment (mmenestr) [reply packages/v4/patternfly-docs/content/design-guidelines/components/text-input-group/text-input-group.md:22]: Would also delete the "here" at the end of the sentence, and just link to the "React demo" or say "can be found in this React demo" ..

**Comment by mcarrano** [comment]
For the text input group, I'm not sure "when to use" is relevant as this is more like a building block that developers can use to build other things. I will make the other updates and push a new version.

**Review by mmenestr — APPROVED** [review]
@mcarrano looks good! merging

**Pull request merged**